src/Surfnet/Stepup/Identity/Entity/SecondFactorCollection.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Stepup\Identity\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+
+final class SecondFactorCollection extends ArrayCollection
+{
+    /**
+     * @return SecondFactor|null
+     */
+    public function getSecondFactorWithHighestLoa()
+    {
+        return array_reduce(
+            $this->toArray(),
+            function (SecondFactor $carry, SecondFactor $item) {
+                return $carry->hasEqualOrHigherLoaComparedTo($item) ? $carry : $item;
+            },
+            $this->first() ?: null
```

Comment:
Won't this break if it is `null`? (see typehint and method invocation in anonymous function)